app/controllers/api_controller/parser/request_adapter.rb

```diff
@@ -6,9 +6,21 @@ def initialize(req, params)
         @params = params
       end
 
+      def to_hash
+        [:method, :action, :fullpath, :url, :base,
+         :path, :prefix, :version, :api_prefix,
+         :collection, :c_suffix, :c_id, :subcollection, :s_id]
+          .each_with_object({}) { |attr, hash| hash[attr] = send(attr) }
+      end
+
       def action
         # for basic HTTP POST, default action is "create" with data being the POST body
-        @action ||= method == :put ? 'edit' : (json_body['action'] || 'create')
```

Comment:
Was this change related to the bug, or an enhancement? Possibly separate PR for this?